spec/Observable-spec.ts

```diff
@@ -125,21 +123,22 @@ describe('Observable', () => {
 
       const results: Array<number | Error> = [];
 
-      return syncObservable.forEach((x) => {
+      syncObservable.forEach((x) => {
         results.push(x);
         if (x === 3) {
           throw expected;
         }
-      }).then(
+      })
+      .catch(err => results.push(err))
+      .then(
         () => {
-          throw new Error('should not be called');
-        },
-        (err) => {
-          results.push(err);
           // Since the consuming code can no longer interfere with the synchronous
           // producer, the remaining results are nexted.
-          expect(results).to.deep.equal([1, 2, 3, 4, expected]);
+          expect(results).to.deep.equal([1, 2, 3, expected]);
```

Comment:
The change to this test's expectation suggests that the behaviour of the previous, non-cancellable implementation was actually a bug - as `next` was called despite an error being effected. The new behaviour makes more sense to me, as an error effected from `next` sees the observer unsubscribe.

Could the comment be removed? It's no longer applicable, as the remaining results are not nexted.

Reply:
Yup. I'll make sure to add the fix to the changelog too.